### ISSUE TITLE
SQL SELECT with GROUP BY and LIMIT returns a broken data

### ISSUE THREAD

**Issue by dimtos** [tarantool/tarantool#4757]
Tarantool version:
2.3.0-236-g6da9d3953

OS version:
Debian 9

Bug description:
A SQL-query 'SELECT * FROM "goods" WHERE "supplier"=22 GROUP BY "id" LIMIT 1' returns the broken data looking as:
rows:
  - [475, 84, '358796подошва рубин36', 19, 451, 22, [36], 1, 5, 2, 3, !!binary 0J7QntCeIGBg0JTRjtC90LAgLSDQAQUCA9lt0J7QntCeIGBg0JTRjtC90LAgLSDQkNCh0KJgYC7QoNCkLCA0MTQwNDQsINCzLtCQ0YHRgtGA0LDRhdCw0L3RjCwg0YPQuy7QodC+0LLQtdGC0Q==, ...
But without LIMIT or without GROUP it's ok.

Steps to reproduce:
goods:
- name: id #primary index
  type: unsigned
  is_nullable: false
- name: product
  type: unsigned
  is_nullable: false
- name: identifier
  type: string
  is_nullable: false
- name: category
  type: unsigned
  is_nullable: false
- name: brand
  type: unsigned
  is_nullable: true
- name: supplier
  type: unsigned
  is_nullable: true
- name: color
  type: array
  is_nullable: true
- name: size
  type: array
  is_nullable: true
- name: count
  type: unsigned
  is_nullable: true
- name: age_group
  type: unsigned
  is_nullable: true
- name: sex
  type: unsigned
  is_nullable: true
- name: package
  type: unsigned
  is_nullable: true
- name: manufacturer
  type: string
  is_nullable: true
- name: material
  type: array
  is_nullable: true
- name: material_inside
  type: array
  is_nullable: true
- name: extra
  type: map
  is_nullable: true
- name: country
  type: unsigned
  is_nullable: true
- name: season
  type: unsigned
  is_nullable: true
- name: unit
  type: unsigned
  is_nullable: true
- name: created
  type: unsigned
  is_nullable: true
- name: price
  type: number
  is_nullable: true

box.execute([[ SELECT * FROM "goods" WHERE "supplier"=22 GROUP BY "id" LIMIT 1 ]])

Optional (but very desirable):
* coredump
* backtrace
* netstat

**Comment by Korablev77** [comment]
SQL does support neither map nor array field types. Even if map/array fields are not explicitly used in query, there's no guarantee that no errors appear during processing. I guess we should restrict using spaces featuring at least one map/array field until these types are integrated with TarantoolSQL. Same concerns https://github.com/tarantool/tarantool/issues/4758

**Comment by ImeevMA** [comment]
Now it's working fine:
```
tarantool> box.execute([[CREATE TABLE t(i INTEGER PRIMARY KEY, a MAP, b ARRAY);]])
---
- row_count: 1
...

tarantool> box.execute([[INSERT INTO t VALUES (1, {'a' : 123}, [1, 17, 'asd']);]])
---
- row_count: 1
...

tarantool> box.execute([[SELECT * FROM t GROUP BY i LIMIT 1;]])
---
- metadata:
  - name: I
    type: integer
  - name: A
    type: map
  - name: B
    type: array
  rows:
  - [1, {'a': 123}, [1, 17, 'asd']]
...
```

Closed by 1364629ec
